DadMakeAFunny/app/src/main/java/com/example/dadmakeafunny/MainActivity.java

```diff
@@ -30,42 +32,51 @@ protected void onCreate(Bundle savedInstanceState) {
         initRecyclerView();
 
         initButtons();
-
     }
 
 
     private void initButtons(){
         Button button = findViewById(R.id.button1);
+
+
         button.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                // Create a toast menu item
                 Toast.makeText(MainActivity.this, "Button 1 pressed", Toast.LENGTH_SHORT).show();
+
+                // Add an item to the recycler view
+                adapter.addAndNotify("Testing Button 1");
+
             }
         });
 
+
+
         button = findViewById(R.id.button2);
         button.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
+                // Create a toast menu item
                 Toast.makeText(MainActivity.this, "Button 2 pressed", Toast.LENGTH_SHORT).show();
+
+                // Add an item to the recycler view
+                adapter.addAndNotify("Testing Button 2");
             }
         });
     }
 
+    // currently doesn't do anything but i feel we might need it for later
     private void initJokes(){
-        jokes.add("This be jokes");
-        jokes.add("This be jokes 1");
-        jokes.add("This be jokes");
-        jokes.add("This be jokes");
-        jokes.add("This be jokes");
+
     }
 
     private void initRecyclerView(){
         Log.d(TAG, "initRecyclerView: started");
 
         RecyclerView recyclerView = findViewById(R.id.Recycler_View);
 
-        RecyclerViewAdapter adapter = new RecyclerViewAdapter(this, jokes);
+        adapter = new RecyclerViewAdapter(this, jokes);

```

Comment:
does jokes here have to be an arraylist of strings ?

Reply:
It would be good to have an ArrayList. Should it be something else?